src/main/java/bot/assembly/function/BotCommandResponderUnit.java

```diff
@@ -0,0 +1,317 @@
+package bot.assembly.function;
+
+import bot.assembly.memory.BotStaticMemoryUnit;
+import bot.assembly.memory.BotDynamicMemoryUnit;
+import bot.assembly.memory.CommandInput;
+import bot.assembly.error.InvalidCommandException;
+import bot.assembly.error.InvalidCommandFormatException;
+import bot.assembly.error.TaskOutOfRangeException;
+import bot.assembly.error.InvalidTaskIndexException;
+
+import bot.assembly.task.Task;
+import bot.assembly.task.Deadline;
+import bot.assembly.task.Event;
+import bot.assembly.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A class that handles the command input of the user
+ */
+public class BotCommandResponderUnit {
+
+    private BotPrinter botPrinter = new BotPrinter();
+    private BotStaticMemoryUnit botStaticMemoryUnit = new BotStaticMemoryUnit();
+    private BotTemporalUnit botTemporalUnit = new BotTemporalUnit();
+    private BotDynamicMemoryUnit botDynamicMemoryUnit = BotDynamicMemoryUnit.getInstance();
+
+    /**
+     * ArrayList as a main data structure to store Task
+     */
+    private List<Task> taskTracker = botDynamicMemoryUnit.taskTracker;
+
+    /**
+     * Constructor
+     */
+    public BotCommandResponderUnit() {}
+
+    /**
+     * A method that tokenizes the command input to 2 parts:
+     * 1. Command Type
+     * 2. Rest of the command
+     * @param input command input
+     * @return [Command Type, Rest of the command]
+     */
+    private String[] tokenize(String input) {
+
+        String[] token = input.split(" ", 2);
+
+        return token;
+    }
+
+    /**
+     * A method that checks the type of command and proceed to break down the command into
+     * 1) Task Title
+     * 2) Task DateTime (format: ISO_LOCAL_DATE_TIME)
+     *
+     * Then proceed to create the respective task and add into the task list
+     * @param input command input
+     * @throws InvalidCommandException if the command input entered is not TODO, EVENT, DEADLINE
+     * @throws InvalidCommandFormatException if the command input entered is in the wrong format
+     */
+    public void addTask(String input) throws InvalidCommandException, InvalidCommandFormatException {
+
+        String[] inputToken = tokenize(input);
+
+        CommandInput taskType = identifyCommand(input);
+
+        switch (taskType) {
+
+            case TODO:
```

Comment:
Should the case clauses have indentation (similarly for the other switch blocks in your code)?